### PR TITLE
Iteration4/logging past activities

### PR DESCRIPTION
**Pull request by JeffShepherd**
## Is this a fix or a feature? ⬇️

feature

## What is the change? ⬇️

Upon completion of the timer, a 'log activity' button appears, and the text on the 'start' button changes to a congratulatory message of 'complete!'. Once the user clicks the 'log activity' button, a card displays on the right section of the screen displaying the category, time, and the users input for 'What would you like to accomplish during this time?'. Additionally, a small indicator changes dynamically to match the color associated with the category.

## Where should the reviewer start? ⬇️

Review all refactored functions: showTimer and checkForErrors. In addition, review new functions displayCard, checkColor, and markComplete.

## How should this be tested? ⬇️

Open index.html in a browser. In addition to all functionality included in prior iterations, test the functionality in the "what is this change?" section listed above.

## Screenshot of functionality ⬇️
Jan-09-2021 16-44-54

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -144,3 +151,23 @@ function resetButtons() {
   removeClass(exerciseButton, "exercise-active");
   exerciseIcon.src = "./assets/exercise.svg";
 }
+
+function displayCard() {
+  addClass(defaultMessage);
+ var color = checkColor(currentActivity.category);
```
Comment (itsnameissteven) [inline main.js]: how about color= currentActivity.category.toLowerCase()

File: main.js [hunk 2]
```diff
@@ -144,3 +151,23 @@ function resetButtons() {
...
+ </div>
+ <div class="card-section">
+   <button class="card-category-indicator ${color}"type="button" name="button"></button>
+ </div>
+ </article>`
+}
+
+function checkColor(category) {
+  return `${category.toLowerCase()}-pipe`
+}
```
Comment (itsnameissteven) [inline main.js]: we can delete this function if we edit 157

File: main.js [hunk 3]
```diff
@@ -77,20 +82,23 @@ function startActivity() {
...
-  if (currentActivity.category === 'Study') {
-    startButton.classList.add('study-active')
-  } else if (currentActivity.category === 'Meditate') {
-    startButton.classList.add('meditate-active')
-  } else if (currentActivity.category === 'Exercise') {
-    startButton.classList.add('exercise-active')
+  changeActivityColor(startButton, 'Study')
+  changeActivityColor(startButton, 'Meditate')
+  changeActivityColor(startButton, 'Exercise')
+}
```
Comment (itsnameissteven) [inline main.js:81]: we could replace 91-31 with:
addClass(startButton, `${currentActivity.category.toLowerCase()}`)

File: main.js [hunk 4]
```diff
@@ -77,20 +82,23 @@ function startActivity() {
...
+
+function changeActivityColor(element, category) {
+  if (currentActivity.category === category) {
+    addClass(element, `${category.toLowerCase()}-active`)
   }
 }
```
Comment (itsnameissteven) [inline main.js]: we can delete this function if we edit 91-93

**Review by itsnameissteven — COMMENTED** [review]
This is looking pretty good. I found some spots that we can slim down  on code and hopefully make the JS more readable.... I also realized after using the disable button for Log-activity that we could use it in other spots in the script to cut out a decent amount of code.  Happy to push what I worked on for refactoring if you want.

File: main.js [hunk 5]
```diff
@@ -77,20 +82,23 @@ function startActivity() {
   showTimer();
 }
 
-// new function
```
Comment (itsnameissteven) [inline main.js:80]: this comment is actually for selectActivityButton function. we could shave it down to: 
selectActivityButton.addEventListener("click", function(event) {
  var button = event.target;
  resetButtons();
  button.disabled = true 
  button.firstElementChild.src = `./assets/${button.value.toLowerCase()}-active.svg`
});

**Comment by JeffShepherd** [comment]
All comments have been been addressed and pushed to this branch.

**Pull request merged**